core/src/main/scala/kafka/admin/AclCommand.scala

```diff
@@ -0,0 +1,241 @@
+    for {
+      principal <- principals
+      operation <- operations
+      host <- hosts
+    } acls += new Acl(principal, permissionType, host, operation)
+
+    acls.toSet
+  }
+
+  private def getHosts(opts: AclCommandOptions, hostOptionSpec: ArgumentAcceptingOptionSpec[String],
+                       principalOptionSpec: ArgumentAcceptingOptionSpec[String]): Set[String] = {
+    if (opts.options.has(hostOptionSpec))
+      opts.options.valueOf(hostOptionSpec).toString.split(delimeter).toSet
+    else if (opts.options.has(principalOptionSpec))
+      Set[String](Acl.WildCardHost)
+    else
+      Set.empty[String]
+  }
+
+  private def getPrincipals(opts: AclCommandOptions, principalOptionSpec: ArgumentAcceptingOptionSpec[String]): Set[KafkaPrincipal] = {
+    if (opts.options.has(principalOptionSpec))
+      opts.options.valueOf(principalOptionSpec).toString.split(delimeter).map(s => KafkaPrincipal.fromString(s)).toSet
+    else
+      Set.empty[KafkaPrincipal]
+  }
+
+  private def getResource(opts: AclCommandOptions): Resource = {
+    val resource: Resource = if (opts.options.has(opts.topicOpt))
+      new Resource(Topic, opts.options.valueOf(opts.topicOpt).toString)
+    else if (opts.options.has(opts.clusterOpt))
+      Resource.ClusterResource
+    else if (opts.options.has(opts.groupOpt))
+      new Resource(ConsumerGroup, opts.options.valueOf(opts.groupOpt).toString)
+    else
+      null
+
+    if (resource == null)
+      CommandLineUtils.printUsageAndDie(opts.parser, "You must provide at least one resource: --topic <topic> or --cluster or --consumer-group <group>")
+
+    resource
+  }
+
+  private def confirmaAction(msg: String): Boolean = {
+    println(msg)
+    val userInput: String = Console.readLine()
+
+    "y".equalsIgnoreCase(userInput)
+  }
+
+  class AclCommandOptions(args: Array[String]) {
+    val parser = new OptionParser
+    val config = parser.accepts("config", "REQUIRED: Path to server.properties file.")
+      .withRequiredArg
+      .describedAs("config")
+      .ofType(classOf[String])
+
+    val topicOpt = parser.accepts("topic", "Topic to which acls should be added or removed.")
+      .withRequiredArg
+      .describedAs("topic")
+      .ofType(classOf[String])
+    val clusterOpt = parser.accepts("cluster", "Add/Remove cluster acls.")
+    val groupOpt = parser.accepts("consumer-group", "Add remove acls for a consumer-group.")
+      .withRequiredArg
+      .describedAs("consumer-group")
+      .ofType(classOf[String])
+
+    val addOpt = parser.accepts("add", "Indicates you are trying to add acls.")
+    val removeOpt = parser.accepts("remove", "Indicates you are trying to remove acls.")
+    val listOpt = parser.accepts("list", "List acls for the specified resource, use --topic <topic> or --consumer-group <group> or --cluster.")
+
+    val operationsOpt = parser.accepts("operations", "Comma separated list of operations, default is All. Valid operation names are: " + nl +
```

Comment:
Instead of using comma separated list, perhaps we should just treat operationsOpt as a list and use options.valuesOf() to parse it. Ditto for allowprincipals, denyprincipals, allowhosts, and denyhosts.

Also, could you list the valid operations in the description?